be/src/storage/rowset/column_reader.h

```diff
@@ -242,10 +241,13 @@ class ColumnReader {
     StarRocksCallOnce<Status> _bitmap_index_once;
     StarRocksCallOnce<Status> _bloomfilter_index_once;
 
+    // Pointer to its father segment, as the column reader
+    // is never released before the end of the parent's life cycle,
+    // so here we just use a normal pointer
+    Segment* _segment;
```

Comment:
1. should be `const Segment`.
2. better to give a default value nullptr.

Although this change will reduce the ColumnReader memory,
this will introduce dependence on Segment. 

And segments depend on ColumnReader now, so this will cause dependence loop.

So we should make Segment usage in ColumnReader as simple as possible.